Rakefile

```diff
@@ -262,6 +263,399 @@ task :xcode => [:dependencies] do
+      end
+    #End namespace xcode-app
+    end
+
+    #XCode-select command line tools must be installed to update dependencies
+    #xcode_select checks the existence of xcode-select on developer's machine, installs if not found
+    namespace :xcode_select do
+      task :check do
+        puts "Checking system for XCode-select"
+        unless command?("xcode-select")
+          Rake::Task["xcode:xcode_select:install"].invoke
+        else
+          puts "XCode-select installed"
+        end
+      end
+
+      task :install do
+        puts "Installing xcode select"
+        sh "xcode-select --install"
+      end
+    #End namesapce :xcode-select
+    end
+  #End namespace xcode
+  end
+
+  #Tools namespace deals with installing developer and OSS tools required to work on WPiOS
+  namespace :tools do
+    task :check_oss => %w[homebrew:check addons:check_oss]
+    task :check_developer => %w[homebrew:check addons:check_developer]
+
+    #Check for Homebrew and install if missing
+    namespace :homebrew do
+      task :check do
+        puts "Checking system for Homebrew"
+        unless command?("brew")
+          Rake::Task["tools:homebrew:install"].invoke
+        else
+          puts "Homebrew installed"
+        end
+      end
+
+      task :install do
+        command = "/bin/bash -c \"$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)\""
+        sh command
+      end
+    #End namespace homebrew
+    end
+
+    #Install required tools to work with WPiOS
+    namespace :addons do
+      #NOTE: hash key = default installed directory on device
+      # hash value = brew install location
+      oss_tools = {"convert" => "imagemagick",
+                  "gs" => "ghostscript",
+      }
+      developer_tools = {"convert" => "imagemagick",
+                        "gs" => "ghostscript",
+                        "sentry-cli" => "getsentry/tools/sentry-cli",
+                        "gpg" => "gpg",
+                        "git-crypt" => "git-crypt",
+      }
+
+      #Check for tool, install if not installed
+      task :check_oss do
+        tool_check(oss_tools)
+      end
+
+      task :check_developer do
+        tool_check(developer_tools)
+      end
+
+      #check machine for if the developer tool is present, if not install
+      def tool_check(hash)
+        hash.each do |key, value|
+          puts "Checking system for #{key}"
+          unless command?(key)
+            tool_install(value)
+          else
+            puts "#{key} found"
+          end
+        end
+      end
+
+      ##install selected developer tool
+      def tool_install(tool)
+        puts "#{tool} not found.  Installing #{tool}"
+        sh "brew install #{tool}"
+      end
+
+    #End namespace addons
+    end
+
+  #End namespace Tools
+  end
+
+  namespace :lint do
+    task :check do
+      if !git_initialized?
+        puts "Initializing git repository"
+        sh "git init", verbose: false
+      end
```

Comment:
Is this check necessary? Is there a scenario in which a user might run this code on the WordPress repo _without_ the `.git` folder set up? 🤔

Reply:
I hadn't put this in originally, but while testing, I setup a completely fresh install of WPiOS  (new OSX, new Xcode, new clone of the WPiOS repo) and the setup failed because I had not yet initialized the git repo on the folder.  So I thought I would build the check in just incase.

I agree that it is really unlikely that a user would run this setup without the .git folder having been created, but it the folder isn't there then the rake setup aborts.  

I do think that the check should be there, because if isn't and installing the swift lint hooks fails, there isn't a lot of indication as to why. But I could add some notes to clarify why the check is there.  What do you think?  Or is there a reason that should be left out that I missed?